src/coreclr/scripts/genEventPipe.py

```diff
@@ -731,6 +746,48 @@ def getMonoEventPipeHelperFileImplPrefix():
     return write_buffer ((const uint8_t *)value, (value_len + 1) * sizeof(*value), buffer, offset, size, fixed_buffer);
 }
 
+bool
+write_buffer_string_utf16_t (
+    const uint8_t *value,
+    size_t value_len,
+    uint8_t **buffer,
+    size_t *offset,
+    size_t *size,
+    bool *fixed_buffer)
+{
+    if (!value)
+        return true;
+
+#if BIGENDIAN
+    bool success = true;
+    const uint8_t *ptr = value;
+    for (size_t i = 0; i <= value_len; ++i) {
+        uint16_t val;
+        memcpy (&val, ptr, sizeof (val));
+        val = ep_rt_val_uint16_t (val);
+        success &= write_buffer ((const uint8_t *)&val, sizeof (val), buffer, offset, size, fixed_buffer);
+        ptr += 2;
+    }
+    return success;
+#else
+    return write_buffer ((const uint8_t *)value, (value_len + 1) * sizeof (*value), buffer, offset, size, fixed_buffer);
+#endif // BIGENDIAN
+}
+
+size_t
+strlen_utf16_t (const uint8_t *value)
```

Comment:
ep_rt_utf16_string_len ?

Reply:
Function `ep_rt_utf16_string_len` expects a pointer to an `ep_char16_t` which is 2 byte aligned and I'm coming in with a potentially 1 byte aligned pointer. Therefore, I came up with function `strlen_utf16_t`.